src/twisted/python/filepath.py

```diff
@@ -38,7 +39,24 @@
     S_IXOTH,
     S_IXUSR,
 )
-from typing import IO, Union, cast
+from typing import (
+    IO,
+    TYPE_CHECKING,
+    AnyStr,
+    Callable,
+    Dict,
```

Comment:
Why `__future__.annotations` _and_ `Dict`, `List`, etc?

Reply:
`__future__.annotations` lets you forward-declare stuff, it doesn't affect builtin types.  Which means that APIs like [`inspect.get_annotations(..., eval_str=True)`](https://docs.python.org/3.11/library/inspect.html#inspect.get_annotations) (and its more ad-hoc predecessors) will work just fine in the presence of `__future__.annotations` on older Python versions, but will *not* work if you start using `list[...]`, `dict[...]`, et. al.

Reply:
(When we drop <3.9, [we can ask pyupgrade to make this transformation for us mechanically](https://github.com/asottile/pyupgrade/tree/67e006e96f7fab6533ab5f831872e72c109a7b28#pep-585-typing-rewrites), and I look forward to doing so.)